### ISSUE TITLE
(Bug) Tally on Panasonic AE-150 Not Responding to VMix Program Variable

### ISSUE THREAD

**Issue by BrianMSimons** [bitfocus/companion#1448]
I grabbed the latest experimental build to play around with since the current stable one doesn't have my PTZ cameras or RP150 controller.  I wanted to use the variable "vmix:mix1_program" for each camera to trigger it's Tally Light and it doesn't work.  I was able to trigger the Tally lights from the RP-150 controller.

At one point I had Camera 1 Tally on from the RP-150 variable. I then changed it to the Vmix variable and the tally stayed on.  I cut to a different camera in Vmix and the Tally went off, as it should.  However it didn't light up the tally on the camera I cut to, and the tally on that camera didn't come back on when I cut back to it.

**Comment by McHauge** [comment]
@BrianMSimons Thanks for the report and for using the PTZ module. I'll be honest I haven't tested the tally part to much since I did a lot of changes, I'll give it a look and see if I can reproduce it here.

Another thing are you sure it's an AE-150 and not an AW-UE150 you have if so I'm quite interested in seeing what it reports when you visit: [your PTZ's IP]/live/camdata.htm

@josephdadams would you mind moving this to the Panasonic-PTZ module instead :)

**Comment by JeffreyDavidsz** [comment]
Put some screenshots of your setup please. Each camera (ipsettings and tally settings) and a screenshot of the vmix settings where you can see the variables. If used this many times with an Atem.

**Comment by BrianMSimons** [comment]
Hey Guys, so yes - the cameras are AW-UE150s. I'm attaching screenshots of the Tally setup for one of the cameras, the current state of the vMix variable, and a photo of the camera....no tally.  When I access the cameras web interface @ [PTZ's IP]/live/camdata.htm I get a 404 error.  I did open the Product Info page on the settings and grabbed an image.

I did get the tally to work if I set the variable based on the camera that's selected on the RP150 controller, but that's not much use to me other than knowing it worked.  

For what it's worth, I just came across Companion this morning while looking for a solution to send Tally from vMix without using NDI.  So I'm completely new to this, but very intrigued given that I've been using Streamdecks, Midas MR32R & MR18s, VMix and now these new cams.  Seems like everything I use can be coordinated with this nifty app.  Thanks.

![Camera Product Info](https://user-images.githubusercontent.com/909111/109054424-d92ca900-76ab-11eb-8d93-5b990639efc3.PNG)

![Tally Off](https://user-images.githubusercontent.com/909111/109054441-ddf15d00-76ab-11eb-8f0c-675ce9a28b20.jpg)

![VMix Current Program Variable Value](https://user-images.githubusercontent.com/909111/109054360-c6b26f80-76ab-11eb-953b-71432b2e1ad0.PNG)

![PTZ 1 Settings](https://user-images.githubusercontent.com/909111/109054326-bc907100-76ab-11eb-80fc-b2eac4768ba8.PNG)

**Comment by McHauge** [comment]
Nice, could you please also add an image about your ue150 instance config page ☺️

**Comment by McHauge** [comment]
Sorry the last picture did not load on my phone 🙃

**Comment by McHauge** [comment]
@BrianMSimons Just tested this with our AW-UE150's and yes there is a bug when using the input number (apparently) BUT it seems to work correctly if you use the input name instead, like this: 
![image](https://user-images.githubusercontent.com/31018035/109063221-36931b00-76e9-11eb-8d21-b2eb489548c2.png)

**Comment by McHauge** [comment]
@BrianMSimons Thanks for the bug report again, I have located the error and pushed a fix, so it should work correctly in new beta builds.

The problem was it was comparing an integer with a variable, which does not end well :)

**Comment by BrianMSimons** [comment]
@McHauge Thank you for fixing it.  I switched over to using the input name from Vmix and it works perfectly.  Even better, it turns the buttons on the RP150 Red for the currently active camera as well.  That should go a long way to helping prevent dumb mistakes.  I think these cameras have both red and green tally lights - that will have to be the next trick.  Make them green when in preview.  I'm so pumped about this app, I can't believe I've never come across it before this morning.  Cheers!

**Comment by McHauge** [comment]
@BrianMSimons Yes from the documentation it would seem that they have green tally too, but I have not been able to turn it on in any way at all, so I have kinda given up trying. after using a lot of time on the tests both in companion, via a controller, the web interface and direct http/tcp copmmands
